giraph-core/src/main/java/org/apache/giraph/writable/kryo/KryoWritableWrapper.java

```diff
@@ -120,4 +126,50 @@ public static Writable wrapIfNeeded(Object object) {
   public static <T> T wrapAndCopy(T object) {
     return WritableUtils.createCopy(new KryoWritableWrapper<>(object)).get();
   }
+
+  /**
+   * Try converting the object to byte array.
+   * @param object Object
+   * @param <T> Type
+   * @return byte array
+   */
+  public static <T> byte [] tryConvertToByteArray(T object) {
+    byte [] arr = null;
+    try {
+      KryoWritableWrapper<T> wrapper =
+              new KryoWritableWrapper<>(object);
+      arr = WritableUtils.toByteArrayUnsafe(wrapper);
+      // Checkstyle exception due to unsafe conversion
+      // CHECKSTYLE: stop IllegalCatch
+    } catch (Exception e) {
```

Comment:
Is there some exception you saw because of which you are putting this, or just as safety? (if latter, it might be better not to do it in these utility methods but from place where you call them)

Reply:
The reason I added was because, the caller could chose any T values, and if the object that was initially converted to byte array is not related to T, then it can throw. But more importantly, there are other callers of this functionality and I don't want to have repeating try-catch blocks. Another option is to add a new utility class that would include the try-catch. Let me know if you think that would be better.